.github/workflows/cms-tests.yml

```
@@ -0,0 +1,45 @@
+name: CMS
+
+on: push
+
+jobs:
+  run-tests:
+    continue-on-error: true
+    runs-on: ubuntu-20.04
+    strategy:
+      matrix:
+        python-version: ['3.8']
+        django-version: ["2.2"]
+        mongodb-version: ['4.0']
+        test_module: [
+            "cms/djangoapps/api/ cms/djangoapps/cms_user_tasks/ cms/djangoapps/course_creators/ cms/djangoapps/export_course_metadata/ cms/djangoapps/maintenance/ cms/djangoapps/models/ cms/djangoapps/pipeline_js/ cms/djangoapps/xblock_config/ cms/envs/ cms/lib/",
+            "cms/djangoapps/contentstore/",
+        ]
+        
+
+    name: ${{ matrix.test_module }}
+    steps:
+      - uses: actions/checkout@v2
+
+      - name: Set up Python ${{ matrix.python-version }}
+        uses: actions/setup-python@v2
+        with:
+          python-version: ${{ matrix.python-version }}
+
+      - name: Install Required System Packages
+        run: sudo apt-get update && sudo apt-get install libxmlsec1-dev
+
+      - name: Start MongoDB
+        uses: supercharge/mongodb-github-action@1.6.0
+        with:
+          mongodb-version: ${{ matrix.mongodb-version }}
+
+      - name: Install requirements
+        run: |
+          sudo pip install --ignore-installed -r requirements/pip.txt
+          sudo pip install --ignore-installed -r requirements/edx/testing.txt
+          sudo pip install "django~=${{ matrix.django-version }}.24"
+      
+      - name: Run tests
+        run: |
+          pytest ${{ matrix.test_module }}
```

I opened this PR #558 to illustrate how to run tests with a retry option, and they fail less -because they were failing at the 1st attempt, now they have 3-. Can we try that here?

```suggestion
      - name: Run tests
        uses: nick-invision/retry@v2
        with:
          timeout_minutes: 45
          max_attempts: 3
          retry_on: error
          command: pytest ${{ matrix.test_module }}
```

Again, this won't prevent tests from failing, but that will be less frequent and nothing that a manual restart job can't solve.

cc @felipemontoya